server/utils/radioMatrixTable.test.ts

```diff
@@ -0,0 +1,96 @@
+import { cell, row, radioMatrixTable } from './radioMatrixTable'
+
+describe('radioMatrixTable', () => {
+  describe('cell', () => {
+    it('returns the markup given the name and preference', () => {
+      expect(cell('name', 'preference')).toMatchStringIgnoringWhitespace(`<td class="govuk-table__cell">
+        <div class="govuk-radios" data-module="govuk-radios">
+            <div class="govuk-radios__item">
+              <input class="govuk-radios__input" id="name-preference" name="name" type="radio" value="preference">
+              <label class="govuk-label govuk-radios__label" for="name-preference"></label>
+            </div>
+          </td>`)
+    })
+
+    it('if passed a check parameter it adds the "checked" tag', () => {
+      expect(cell('name', 'preference', true)).toMatchStringIgnoringWhitespace(`<td class="govuk-table__cell">
+        <div class="govuk-radios" data-module="govuk-radios">
+            <div class="govuk-radios__item">
+              <input class="govuk-radios__input" id="name-preference" name="name" type="radio" value="preference" checked >
+              <label class="govuk-label govuk-radios__label" for="name-preference"></label>
+            </div>
+          </td>`)
+    })
+  })
+
+  describe('row', () => {
+    it('returns the markup given the name and preferences', () => {
+      expect(row('test', ['foo', 'bar'], 'foo')).toMatchStringIgnoringWhitespace(`<tr>
+      <th class="govuk-table__cell" scope="row">Test</td>
+        <td class="govuk-table__cell">
+      <div class="govuk-radios" data-module="govuk-radios">
+              <div class="govuk-radios__item">
+                <input class="govuk-radios__input" id="test-foo" name="test" type="radio" value="foo" checked>
+                <label class="govuk-label govuk-radios__label" for="test-foo"></label>
+              </div>
+            </td>
+      <td class="govuk-table__cell">
+    <div class="govuk-radios" data-module="govuk-radios">
+            <div class="govuk-radios__item">
+              <input class="govuk-radios__input" id="test-bar" name="test" type="radio" value="bar">
+              <label class="govuk-label govuk-radios__label" for="test-bar"></label>
+            </div>
+          </td>
+  </tr>`)
+    })
+  })
+
+  describe('radioMatrixTable', () => {
```

Comment:
If you're testing the `row` and `cell` methods in isolation, then you can probably slim this test down by just checking the `row` and `cell` methods are being called at the right place with the right args. Something like:

```typescript
     expect(
        radioMatrixTable(['heading one', 'heading two'], ['valueOne', 'valueTwo'], ['preferenceOne', 'preferenceTwo'], {
          valueOne: 'preferenceTwo',
        }),
      ).toMatchStringIgnoringWhitespace(
        `<table class="govuk-table">
        <thead class="govuk-table__head">
          <tr class="govuk-table__row">
          <th class="govuk-table__header" scope="col">heading one</th><th class="govuk-table__header" scope="col">heading two</th>
          </tr>
        </thead>
        row('valueOne', ['preferenceOne', 'preferenceTwo'], undefined)
        row('valueTwo', ['preferenceOne', 'preferenceTwo'], 'preferenceTwo')
      </table>`,
      )
```

Reply:
Maybe even adding a `tableHeadings` method would help simplify things even further?

Reply:
I agree with this approach but last time I tried it was hard to mock functions in the same file as the test subject but I'll have another go as I haven't tried it recently.

Reply:
I wouldn't worry so much about mocking, just add the literal call to the function in the expectation

Reply:
Apologies but I'm not following, do you mean: 
1. `expect(row).toHaveBeenCalledWith(foo, bar)`
2. `expect(row(arg1, arg2, arg3).toBe('...')`
3. something else?

Reply:
By interpolating the literal function calls in the string your expecting. I realised my suggestion had bad code in it, so haved edited above 👍

Reply:
I see!!! sorry missed your example the first time around 🤦‍♂️